app/src/main/java/com/bogdan/codeforceswatcher/activity/AddUserActivity.kt

```diff
@@ -49,15 +49,22 @@ class AddUserActivity : AppCompatActivity(), OnClickListener {
                         val localUser = response.body()!!.result.firstOrNull()!!
                         ratingCall.enqueue(object : Callback<RatingChangeResponse> {
                             override fun onResponse(call: Call<RatingChangeResponse>, response: Response<RatingChangeResponse>) {
+                                progressBar.visibility = View.INVISIBLE
                                 if (response.isSuccessful) {
                                     localUser.ratingChanges = response.body()!!.result
-                                    CwApp.app.userDao.insert(localUser)
-                                    progressBar.visibility = View.INVISIBLE
+                                    if (CwApp.app.userDao.getAll().find { it.handle == localUser.handle } != null) {
+                                        Toast.makeText(applicationContext, getString(R.string.user_already_added), Toast.LENGTH_SHORT).show()
+                                        return
```

Comment:
It would be more logical to not use return here and have finish only when user added.